### ISSUE TITLE
Add ability to specify hosts on the command line via netmask

### ISSUE THREAD

**Issue by GoogleCodeExporter** [brockgr/csshx#26]
```
It would be really nice to have the capability on the command line to do
something along the lines of:
csshx --login elubow 192.168.1.12/29

This would open up everything in that netmask.
```

Original issue reported on code.google.com by `eric.lu...@gmail.com` on 27 Mar 2010 at 1:55

**Comment by GoogleCodeExporter** [comment]
```
Yes - nice idea. I'll look at how difficult it is to implement.

Wonder if I should have some safety checks in case someone tries to csshX 
0.0.0.0/0

Maybe /24 or more only?
```

Original comment by `gavin.brock` on 27 Mar 2010 at 9:13
- Changed state: **Accepted**
- Added labels: **Type-Enhancement**
- Removed labels: **Type-Defect**

**Comment by GoogleCodeExporter** [comment]
```
I didn't realize this was in Perl (sorry should have looked).  I will take a 
swing at 
implementing this in short order and send you a patch.  But agreed that a limit 
should 
be set on the number of hosts that connections can be opened to.  This can also 
prevent DoSing yourself if you inadvertently type 192.168.1.1/2 in stead of /28 
or 
something like that.
```

Original comment by `eric.lu...@gmail.com` on 27 Mar 2010 at 3:33

**Comment by GoogleCodeExporter** [comment]
```
I have attached a patch which will allow for either 192.168.1.1/XX or 
192.168.1.1/255.255.255.XXX.  There is some very rudimentary error checking.  I 
didn't 
want to force the user to have to install Perl modules, so I borrowed code from 
some 
other CPAN modules (attributing credit where necessary) to make this happen.

The maximum allowable number of sessions is 255.  Additionally I have added a 
command line switch to allow someone to override this behavior.
```

Original comment by `eric.lu...@gmail.com` on 28 Mar 2010 at 11:03

Attachments:
- [patch64](https://storage.googleapis.com/google-code-attachments/csshx/issue-26/comment-3/patch64)

**Comment by GoogleCodeExporter** [comment]
```
Eric

Thank you - nothing like sending a patch to prompt action :-)

I took a mixture of your patch and some other code I had lying around (slightly 
shorter netmask parsing code) 
and added it.

While I was in the mood, I also added support for ranges: 192.168.1.[12-24]

This naturally led to fun stuff like: db[pri,sec][1-9][a-f,z]

You can grab this from SVN at http://csshx.googlecode.com/svn/trunk/csshX . Let 
me know if I seem to have 
broken anything else.
```

Original comment by `gavin.brock` on 29 Mar 2010 at 3:00
- Changed state: **Started**

**Comment by GoogleCodeExporter** [comment]
```
It's almost everything I expected (great work in combining the code).  The out 
of /31 should be 2 hosts yet its only 1:
elubow@beacon trunk$ ./csshX --login elubow baz@192.168.1.1/31 
elubow@209.131.125.101/30  
foobar@209.131.125.113/255.255.255.252
$VAR1 = [
          'baz@192.168.1.1',
          'elubow@209.131.125.101',
          'elubow@209.131.125.102',
          'elubow@209.131.125.103',
          'foobar@209.131.125.113',
          'foobar@209.131.125.114',
          'foobar@209.131.125.115'
          ];

According to ipcalc, it should be 192.168.1.0 and 192.168.1.1 (yes I cheat with 
ipcalc):
elubow@beacon ipcalc-0.41$ ./ipcalc 192.168.1.1/31
Address:   192.168.1.1          11000000.10101000.00000001.0000000 1
Netmask:   255.255.255.254 = 31 11111111.11111111.11111111.1111111 0
Wildcard:  0.0.0.1              00000000.00000000.00000000.0000000 1
=>
Network:   192.168.1.0/31       11000000.10101000.00000001.0000000 0
HostMin:   192.168.1.0          11000000.10101000.00000001.0000000 0
HostMax:   192.168.1.1          11000000.10101000.00000001.0000000 1
Hosts/Net: 2                     Class C, Private Internet, PtP Link RFC 3021

Otherwise everything seems to be working just fine.  Let me know when the final 
version is released and I'll add another post 
in my blog (http://eric.lubow.org/).
```

Original comment by `eric.lu...@gmail.com` on 29 Mar 2010 at 3:01

**Comment by GoogleCodeExporter** [comment]
```
Eric - thanks for the blog post - much appreciated!

Regarding the missing IP - it's a 'feature'. I count up from the IP address 
specified, rather than the network 
address (HostMin):

i.e. 
192.168.1.1/31 = 192.168.1.1
192.168.1.0/31 = 192.168.1.0 and 192.168.1.1

I assumed this would give the user a little more control. If you really wanted 
the whole network, you would 
specify the network address to start.

That said, it's only one more line to change the behavior so if you think this 
is really illogical, I can swap it.

```

Original comment by `gavin.brock` on 29 Mar 2010 at 11:23

**Comment by GoogleCodeExporter** [comment]
```
No worries on the blog post, I try to keep up with things that are useful and 
this has 
been REALLY useful to me lately.  I'll follow it up with another blog post when 
you 
release this next version (since I did contribute and all).

Nah, no need to change the behavior.  As long as its spelled out in the pod 
documentation as expected behavior.  
```

Original comment by `eric.lu...@gmail.com` on 30 Mar 2010 at 4:31

**Comment by GoogleCodeExporter** [comment]
```
Released in 0.64: http://gavcode.wordpress.com/2010/03/30/csshx-0-64/

```

Original comment by `gavin.brock` on 30 Mar 2010 at 2:05
- Changed state: **Done**
